crates/notion-core/src/version/mod.rs

```diff
@@ -70,23 +68,14 @@ impl FromStr for VersionSpec {
     }
 }
 
-#[derive(Debug, Fail, NotionFail)]
-#[fail(display = "{}", error)]
-#[notion_fail(code = "NoVersionMatch")]
-pub(crate) struct VersionParseError {
-    pub(crate) error: String,
-}
-
-impl VersionParseError {
-    fn from_req_parse_error(error: &ReqParseError) -> Self {
-        VersionParseError {
-            error: error.to_string(),
-        }
+fn from_req_parse_error(error: &ReqParseError) -> ErrorDetails {
```

Comment:
I don't _love_ top-level functions with "from_xxx" names, and this is related to the other naming suggestions I made. But in this case you have two different semver parse errors going into one `ErrorDetails` variant, so you can't just have one name. Maybe just `version_req_parse_error` and `version_parse_error`.